### ISSUE TITLE
Question about highlighting code fragments

### ISSUE THREAD

**Issue by nzubair** [PowerShell/PowerShell-Docs#1018]
Issue Details
-----------------

Can you please advise what's the proper way to highlight code fragments? 

I've seen this:

> \```
> PS C:\>Get-Process
> \```
> 

and 

> \```powershell
> PS C:\>Get-Process
> \```

I've seen some comments (can't recall where) that due to `PS C:\>` part, this is not a powershell code fragment. Therefore it should not be highlighted by **\```powershell`**, rather by simple three backticks. 

Can you advise what is preferred? 

Version(s) of document impacted
------------------------------
- [x] Impacts 5.1 document
- [x] Impacts 5.0 document
- [x] Impacts 4.0 document
- [x] Impacts 3.0 document

**Comment by wsmelton** [comment]
Markdown formatting allows you to use the `powershell` keyword, it won't necessarily highlight the prompt section but the remaining code (e.g. Get-Process) would be highlighted appropriately.

```powershell
PS C:> Get-Process
```

```
PS C:> Get-Process
```

It works the same way when you post code in issues on GitHub.

**Comment by nzubair** [comment]
I understand that. My question was whether or not the PowerShell-Docs team has a preference.

**Comment by nzubair** [comment]
Found the comment. It was @vors who made this comment: https://github.com/PowerShell/PowerShell-Docs/pull/895#discussion_r93661168

**Comment by HemantMahawar** [comment]
IMHO, once you tag the code with block with powershell (` ```powershell`) , there is no need for `PS C:>` prompt as it does appropriate highlighting of PowerShell code. Also, it makes copy-paste easier without the need to copy/remove the PowerShell prompt.

**Comment by joeyaiello** [comment]
I agree with @HemantMahawar, but unfortunately that doesn't work for when cmdlet reference examples get turned into MAML (which is why we probably shouldn't remove them).

**Comment by HemantMahawar** [comment]
@juanpablojofre @zjalexander @eslesar @joeyaiello Do any of you have additional thought on this issue?

**Comment by HemantMahawar** [comment]
Good point @joeyaiello about maml help /cc: @zjalexander @juanpablojofre @eslesar

**Comment by zjalexander** [comment]
@eslesar and @juanpablojofre and I have all discussed this, the consensus is:

Do NOT use PS C:>

DO use ```powershell. 

Many of our docs today do not conform to this since they were ported forward. Many of our docs will probably not conform to this in the future . 

I also discussed the MAML issue with @joeyaiello and we can resolve this with "make sure to put an extra line break between a command and command output ". 

This raises a few other issues - hey, don't we need a style guide (#15) ? can we get rid of extraneous linebreaks that have been created through the years (#1025)? 


@nzubair - does this answer your question fully?

**Comment by nzubair** [comment]
Yes it does. Thank you.

**Comment by nzubair** [comment]
Was playing around with this and what I've found that removing PS C:> seems to make the example hard to read. 

```powershell
PS C:> "abc" -eq "abc"
True
```
vs.

```powershell
"abc" -eq "abc"

True
```

**Comment by exchange12rocks** [comment]
Maybe, the best option would be to remove the prompt from a multi-line commands while leaving it at single-line examples.

**Comment by wsmelton** [comment]
Well formatting wise so folks can follow what is output and what is code it would be best to just break them up for the examples. If you kept the "PS C:>" it was a better indicator of what command was actually being shown.

Code:
```powershell
"abc" -eq "abc"
```
Output:
```
true
```

**Comment by nzubair** [comment]
That's sensible. .Net folks seem to be doing something similar (https://docs.microsoft.com/en-us/dotnet/articles/csharp/tour-of-csharp/index). However, they are doing some post-processing when it's published live, as opposed to doing everything in MarkDown.

That being said, there are quite a few example-heavy documents, where this format will not work. It becomes confusing and unreadable quickly, especially if there is no additional formatting employed. 

Just looking at MarkDown docs, take a look at the following two docs (excerpts from about comparison operators)

Current formatting: https://github.com/nzubair/TestRepo/blob/master/test1.md
New formatting: https://github.com/nzubair/TestRepo/blob/master/test2.md

**Comment by exchange12rocks** [comment]
.Net folks do not include output as a real console output - they just insert it in comments - it makes their life easier in a way..

**Comment by nzubair** [comment]
Not in the page I linked above. They're actually using 

> \```console
> Hello, World
> \```

**Comment by zjalexander** [comment]
@nzubair test2.md would throw a platyps error with the current schema because it only expects 1 version block: https://github.com/PowerShell/platyPS/blob/master/platyPS.schema.md#version-200

related: https://github.com/PowerShell/platyPS/issues/243

anyway, thanks for all the feedback everyone, i'm going to round up some of my team members and see how they feel.

**Comment by nzubair** [comment]
test2.md was @wsmelton's suggestion. I just wanted to see how it would come out.

I've been working on some of the about_* topics. However, I will hold off until there is some clarity on this. 

Thank you.

**Comment by zjalexander** [comment]
One idea @joeyaiello had is to use comment blocks for output to prevent weird syntax highlighting issues:

```powershell
Get-Variable $true
#Name                           Value
#----                           -----
#true                           True
```

**Comment by rjmholt** [comment]
Another thing I've seen -- which very probably can't be done with pure markdown, but might be worth mentioning anyway -- is using the CSS `before` selector for shell prompts: [an example from GitHub](https://help.github.com/articles/checking-for-existing-ssh-keys/) shows a bash prompt with a `$` character, but it can't be selected.

**Comment by zjalexander** [comment]
I believe this issue was resolved by #1095, please open a new issue if that's not accurate.
